### PR TITLE
SASL Mechanism: AWS MSK IAM (making requested edits)

### PR DESCRIPTION
**Pull request by dominicbarnes**
This builds on #763 and adds a commit which makes the changes I requested on the latest of that PR. My intention is to push this commit over to paxosglobal's fork, but I couldn't figure out how to open my PR to reflect that intent. As such, I'm opening the PR directly against main to see if my fixes bring everything into line.

This PR duplicates #763 (and will be merged in it's place)

### CODE REVIEW

**Comment by dominicbarnes** [comment]
@cmaher I've made the changes I requested for your PR #763 here, while preserving the commit history that includes your contribution. If possible, I'd like to push my commit to your branch so we can merge the original PR, but I can also just merge from here if that becomes tricky.

File: sasl/aws_msk_iam/go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module github.com/segmentio/kafka-go/sasl/aws_msk_iam
+
+go 1.15
+
+require (
+	github.com/aws/aws-sdk-go v1.41.3
+	github.com/segmentio/kafka-go v0.4.23
```
Comment (cmaher) [inline sasl/aws_msk_iam/go.mod:7]: I think this still needs to refer to a kafka-go version after the sasl changes have been merged? I also have no idea how the replace statement will work if this is imported somewhere else.

**Comment by cmaher** [comment]
changes look good to me!

File: sasl/aws_msk_iam/go.mod (hunk 1 above)
Comment (dominicbarnes) [reply sasl/aws_msk_iam/go.mod:7]: Ah yeah, good catch. I'll remove that replace statement, then update the version here. I'm not 100% sure how we'll make sure they stay in sync, but we'll cross that bridge after merge.

**Comment by dominicbarnes** [comment]
I think we're all clear now that I tagged v0.4.24 (which includes the SASL Metadata)

**Pull request merged**